AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/AWSCognitoAuthPlugin+Configure.swift

```diff
@@ -217,12 +220,16 @@ extension AWSCognitoAuthPlugin {
                                       cognitoIdentityFactory: makeIdentityClient)
     }
 
-    private func credentialStoreEnvironment(authConfiguration: AuthConfiguration) -> CredentialEnvironment {
+    private func credentialStoreEnvironment(authConfiguration: AuthConfiguration, accessGroup: String? = nil) -> CredentialEnvironment {
         CredentialEnvironment(
             authConfiguration: authConfiguration,
             credentialStoreEnvironment: BasicCredentialStoreEnvironment(
-                amplifyCredentialStoreFactory: makeCredentialStore,
-                legacyKeychainStoreFactory: makeLegacyKeychainStore(service:)
+                amplifyCredentialStoreFactory: {
+                    self.makeCredentialStore(accessGroup: accessGroup)
+                },
+                legacyKeychainStoreFactory:  { service in
+                    self.makeLegacyKeychainStore(service: service, accessGroup: accessGroup)
+                }
```

Comment:
We should not be adding `accessGroup`to the legacy keychain. This would cause migration to fail from V1 to V2.